src/gpuhunt/providers/gcp.py

```diff
@@ -232,3 +262,266 @@ def filter(cls, offers: List[RawCatalogItem]) -> List[RawCatalogItem]:
             )
             or (i.gpu_name and i.gpu_name not in ["K80", "P4"])
         ]
+
+
+def get_tpu_offers(project_id: str) -> List[RawCatalogItem]:
+    logging.info("Fetching tpu offers")
+    raw_catalog_items: List[RawCatalogItem] = []
+    catalog_items: List[dict] = get_catalog_items(project_id)
+    filtered_catalog_items = [item for item in catalog_items if item["price"] is not None]
+    for item in filtered_catalog_items:
+        on_demand_item = RawCatalogItem(
+            instance_name=item["instance_name"],
+            location=item["location"],
+            price=item["price"],
+            cpu=0,
+            memory=0,
+            gpu_count=1,
+            gpu_name=f'tpu-{item["instance_name"]}',
+            gpu_memory=0,
+            spot=False,
+            disk_size=None,
+        )
+        raw_catalog_items.append(on_demand_item)
+        if item["spot"]:
+            spot_item = copy.deepcopy(on_demand_item)
+            spot_item.price = item["spot"]
+            spot_item.spot = True
+            raw_catalog_items.append(spot_item)
+    return raw_catalog_items
+
+
+def get_catalog_items(project_id: str) -> List[dict]:
+    tpu_prices: List[dict] = get_tpu_prices()
+    configs: List[dict] = get_tpu_configs(project_id)
+    for config in configs:
+        instance_name = config["instance_name"]
+        location = config["location"].rsplit("-", 1)[0]  # Remove the part after the last '-'
+        no_of_chips = config["no_of_chips"]
+        tpu_version, no_of_cores = instance_name.rsplit("-", 1)
+        no_of_cores = int(no_of_cores)
+        if any(tpu_version == x for x in ["v5litepod", "v5p"]):
```

Comment:
Maybe `if tpu_version in ["v5litepod", "v5p"]`?

Reply:
Thank for suggestion